### PR TITLE
refactor: Add comprehensive test coverage for CLI and plugins

### PR DESCRIPTION
**Pull request by rolfedh**
This PR adds comprehensive test coverage for the refactored CLI and plugin infrastructure from previous PRs.

## Test Coverage Added

- **`test_cli.py`**: Complete CLI test suite (192 lines, 13 tests)
  - CLI argument parsing and validation 
  - Plugin discovery and execution
  - Error handling for broken plugins
  - Help message functionality
  - Mock-based testing for safe execution

- **`test_EntityReference.py`**: Enhanced plugin tests (+120 lines, 7 tests)
  - Entity replacement functionality
  - Fixture-based testing with graceful degradation  
  - Edge case handling
  - Error validation with proper warnings

## Quality Assurance
- **20 total tests passing**
- All tests use proper mocking to avoid side effects
- Graceful handling of missing test fixtures (warns but continues)
- Tests validate functionality from PRs #1-3
- Maintains backward compatibility with existing test utilities

## Stats
- **297 lines added** across 2 test files
- **0 functional code changes** - pure testing infrastructure
- **20/20 tests passing** ✅

**Dependencies:** Tests the refactored code from PRs #1-3

### CODE REVIEW

File: tests/test_cli.py [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+    def test_plugin_loading_with_missing_function(self):
+        """Test handling of plugins missing register_subcommand function."""
+        with patch('sys.argv', ['toolkit']), \
+             patch('sys.stderr', new_callable=StringIO) as mock_stderr, \
+             patch('asciidoc_dita_toolkit.asciidoc_dita.toolkit.discover_plugins') as mock_discover, \
+             patch('importlib.import_module') as mock_import:
+            
+            mock_discover.return_value = ['test_plugin']
+            mock_module = MagicMock()
+            del mock_module.register_subcommand  # Remove the function
```
[the remark this answers is not in the document]
Comment (rolfedh) [reply tests/test_cli.py:63]: The test is working correctly. This is a false positive.
